tardis/io/atomic.py

```diff
@@ -310,149 +145,69 @@ def from_hdf5(cls, fname=None):
+        with pd.HDFStore(fname) as store:
+            for name in cls.hdf_names:
+                try:
+                    dataframes[name] = store[name]
+                except KeyError:
+                    nonavailable.append(name)
 
-        if 'macro_atom_data' in h5_datasets:
-            macro_atom_data = read_macro_atom_data(fname)
-        else:
-            macro_atom_data = None
+            atom_data = cls(**dataframes)
 
-        if 'zeta_data' in h5_datasets:
-            zeta_data = read_zeta_data(fname)
-        else:
-            zeta_data = None
+            try:
+                atom_data.uuid1 = store.root._v_attrs['uuid1']
+            except KeyError:
+                atom_data.uuid1 = None
 
-        if 'collision_data' in h5_datasets:
-            collision_data, collision_data_temperatures = read_collision_data(fname)
-        else:
-            collision_data, collision_data_temperatures = (None, None)
+            try:
+                atom_data.md5 = store.root._v_attrs['md5']
+            except KeyError:
+                atom_data.md5 = None
 
-        if 'synpp_refs' in h5_datasets:
-            synpp_refs = read_synpp_refs(fname)
-        else:
-            synpp_refs = None
+            try:
+                atom_data.version = store.root._v_attrs['database_version']
+            except KeyError:
+                atom_data.version = None
 
-        if 'ion_cx_data' in h5_datasets and 'ion_cx_data' in h5_datasets:
-            ion_cx_data = read_ion_cx_data(fname)
-        else:
-            ion_cx_data = None
-
-        atom_data = cls(atom_data=atom_data, ionization_data=ionization_data, levels_data=levels_data,
-                        lines_data=lines_data, macro_atom_data=macro_atom_data, zeta_data=zeta_data,
-                        collision_data=(collision_data, collision_data_temperatures), synpp_refs=synpp_refs,
-                        ion_cx_data=ion_cx_data)
+            # ToDo: strore data sources as attributes in carsus
 
-        with h5py.File(fname, 'r') as h5_file:
-            atom_data.uuid1 = h5_file.attrs['uuid1']
-            atom_data.md5 = h5_file.attrs['md5']
-            atom_data.version = h5_file.attrs.get('database_version', None)
+            logger.info(
+                "Read Atom Data with UUID={0} and MD5={1}. "
+                "Not found: {2}".format(atom_data.uuid1, atom_data.md5, ", ".join(nonavailable))
+            )
 
-            if atom_data.version is not None:
-                atom_data.data_sources = pickle.loads(h5_file.attrs['data_sources'])
-
-            logger.info('Read Atom Data with UUID=%s and MD5=%s', atom_data.uuid1, atom_data.md5)
 
         return atom_data
 
-    def __init__(self, atom_data, ionization_data, levels_data, lines_data, macro_atom_data=None, zeta_data=None,
-                 collision_data=None, synpp_refs=None, ion_cx_data=None):
+    def __init__(self, atom_data, ionization_data, levels=None, lines=None,
```

Comment:
Should any of the frames be mandatory to instantiate `AtomData`? Currently, `atom_data` and `ionization_data` are positional arguments so they are mandatory, but that is not sensible (at least `levels` and `lines` also should be positional args then)